src/idom/utils.py

```diff
@@ -56,6 +62,92 @@ def __repr__(self) -> str:
         return f"{type(self).__name__}({current})"
 
 
+def vdom_to_html(value: str | VdomDict, indent: int = 0, depth: int = 0) -> str:
+    """Convert a VDOM dictionary into an HTML string
+
+    Only the following keys are translated to HTML:
+
+    - ``tagName``
+    - ``attributes``
+    - ``children`` (must be strings or more VDOM dicts)
+    """
+    if indent:
```

Comment:
I don't see a major reason to do indentation within this utility.

If the user really wanted things looking pretty, it's best done via a HTML beautifier like `html5print` or `Prettier`, which would give them way more configuration options.